lib/include/TransportDescriptor.hpp

```diff
@@ -0,0 +1,2033 @@
+
+/**
+ * @brief Parse XML file and print the specific TCP Transport Descriptor listening ports collection.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string XML section containing the list of the listening ports in the TCP Transport Descriptor.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        listening ports element has not been set in the profile.
+ */
+std::string print_listening_ports(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Parse XML file and print the TCP Transport Descriptor specific listening port element.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return uint16_t TCP Transport descriptor specific listening port.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        list does not contain any element in index position.
+ */
+uint16_t print_listening_ports(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id,
+        int32_t index);
+
+/**
+ * @brief Parse XML file and print the specific TCP Transport Descriptor TLS configuration.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string XML section containing the specific TCP Transport Descriptor TLS configuration.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        TLS configuration element has not been set in the profile.
+ */
+std::string print_tls(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Parse XML file and print the specific TCP Transport Descriptor TLS password.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string Specific TCP Transport Descriptor TLS password.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        TLS password element has not been set in the profile.
+ */
+std::string print_tls_password(
```

Comment:
In this case that it is only one complex element within the TransportDescriptor, and the complex element name is not that lengthy, we agreed to keep the name in the method instead of creating a new namespace. @EduPonz, what do you think?

Reply:
That is what we agreed, yes